### PR TITLE
Issue 3 change updated to optional in api entry

### LINKED ISSUE #3: "updated" field not always available in ApiEntry (part of CollectionFeed)
First of all thanks for this package. This looks great. Although it is currently focused on LPDAAC datasets, I think it can be easily adapated to datasets from NSIDC DAAC (e.g., MOD10A1 snow cover) as this dataaset is also available within CMR and download authentication is the same. 

I ran into an issue while search for the MOD10A1 (version 061):
```
collections = collection_client.query(short_name="MOD10A1", version="061")

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/venv/lib/python3.8/site-packages/modis_tools/decorators.py", line 20, in wrapped
    result = func(*fargs, **fkwargs, **{reqarg_name: processed})
  File "/home/ubuntu/venv/lib/python3.8/site-packages/modis_tools/resources.py", line 32, in query
    collection_feed = CollectionFeed(**resp.json()["feed"])
  File "pydantic/main.py", line 331, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for CollectionFeed
entry -> 0 -> updated
  field required (type=value_error.missing)
```

You can have a look at the URL requested here: https://cmr.earthdata.nasa.gov/search/collections.json?doi=10.5067%2FMODIS%2FMOD10A1.061

This is related to the ApiEntry base model, which includes the `updated` property as a requirement: 
https://github.com/fraymio/modis-tools/blob/187a2519b4c5fa91a098b098e3eb5f9b8c2a16ef/modis_tools/models.py#L22-L31

It seems to be that this field is not always available. Would it be possible to declare this property as optional?

### PR DESCRIPTION
**Pull request by ShengpeiWang**
## Description

- [made updated optional in ApiEntry](https://github.com/fraymio/modis-tools/commit/f6bc2c2ab05213ec96851c6ba223dcbc031c7ca6)
- [add National Snow and Ice Data Center as a datasource](https://github.com/fraymio/modis-tools/commit/4176dc32d68eef8eba5d91e2fa5e5144889c9daa)
- reverted [898fa80](https://github.com/fraymio/modis-tools/pull/9/commits/898fa80602723c9e26a23368f3dc6f522ef58ed9) in [b97ed66](https://github.com/fraymio/modis-tools/pull/9/commits/b97ed66854e6cfb632872e5edde80daebc5c79f7). The reason for 898fa80: to allow for the successful download of non-modis data. since the get request response headers don't contain "Location". I noted in the docstring that this flag is experimental, since I did not test it using any datasource other than NSIDC

Closes #3 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [x] pytest
- [x] tested that local downloads succeed with the README example: "MOD13A1"
- [x] tested that local downloads succeed with the example from the issue #3 : "MOD10A1"

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Next Steps
- [ ] Assign a reviewer based on the [code owner document](https://github.com/fraymio/modis_tools/blob/main/.github/CODEOWNERS).

- [ ] Once your review is approved, merge and delete the feature branch

On behalf of the Modis Tools Dev Team, thank you for your hard work! ✨

### CODE REVIEW

**Comment by ShengpeiWang** [comment]
Request error during data download as of [898fa80](https://github.com/fraymio/modis-tools/pull/9/commits/898fa80602723c9e26a23368f3dc6f522ef58ed9). @jtanwk and I figured out that the issue: the download_cookies have different data structures for the modis vs the nsidc data. We got successful data download from both sources once we changed the has_download_cookies method by removing the lines: `data = cookies["DATA"]
        assert data.domain == URLs.RESOURCE.value`
Need to check with original contributors and other code owners about how to proceed with the fix.

**Comment by jtanwk** [comment]
Additional note: cookie name for NSIDC is "CisforCookie_OPS" whereas it is "DATA" for LP DAAC-sourced data.

Extra documentation probably required for future updates if we want to add other DAACs/data sources.

File: modis_tools/auth.py [hunk 1]
```diff
@@ -54,9 +54,17 @@ def has_download_cookies(session):
...
+            # specific to the cookies for LP DAAC source
+            data = cookies["DATA"]
+            assert data.domain == URLs.RESOURCE.value
+        elif "CIsForCookie_OPS" in cookies:
+            data = cookies["CIsForCookie_OPS"]
+            # specific to the cookies for NSIDC DAAC source
+            assert data.domain == URLs.NSIDC_RESOURCE.value
+        else:
+            # print(cookies)
+            raise KeyError("resource domain not found in cookies")
```
Comment (jtanwk) [inline modis_tools/auth.py:67]: Maybe this is a good place to add a prompt for the user like:
```suggestion
            raise KeyError("Data source not recognized. Please open an issue informing us of the desired data source.")
```

File: modis_tools/auth.py [hunk 2]
```diff
@@ -54,9 +54,17 @@ def has_download_cookies(session):
...
+        if "DATA" in cookies:
+            # specific to the cookies for LP DAAC source
+            data = cookies["DATA"]
+            assert data.domain == URLs.RESOURCE.value
+        elif "CIsForCookie_OPS" in cookies:
+            data = cookies["CIsForCookie_OPS"]
+            # specific to the cookies for NSIDC DAAC source
+            assert data.domain == URLs.NSIDC_RESOURCE.value
+        else:
+            # print(cookies)
```
Comment (cearlefraym) [inline modis_tools/auth.py:66]: could remove the commented print here

**Comment by cearlefraym** [comment]
MOD10A1 download works now. 
LGTM

**Pull request merged**